addnewcitydialog.cpp

```diff
@@ -0,0 +1,82 @@
+#include "addnewcitydialog.h"
+#include "ui_addnewcitydialog.h"
+
+#include <QIcon>
+
+AddNewCityDialog::AddNewCityDialog(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::AddNewCityDialog),
+    m_messageBoxInfo{new QMessageBox(this)}
+{
+    ui->setupUi(this);
+}
+
+AddNewCityDialog::~AddNewCityDialog()
+{
+    delete ui;
+}
+
+void AddNewCityDialog::on_save_cancel_buttonBox_accepted()
+{
+    if (ui->inputContinentComboBox->currentText() == "Select continent"
```

Comment:
Instead of these checks better repeating the logic from DataLoader::getDataFromLine